[SPARK-10169] [SQL] [BRANCH-1.2] Partial aggregation's plan is wrong when a grouping expression is used as an argument of the aggregate fucntion

yhuai opened the pull request:
https://issues.apache.org/jira/browse/SPARK-10169

yhuai committed:
Use transformDown.

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #41420 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41420/consoleFull) for   PR 8381 at commit [`c4c1b28`](https://github.com/apache/spark/commit/c4c1b28685849b460ea343d8db3c8becce0d6de2).
 * This patch merges cleanly.

SparkQA commented:
[Test build #41420 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41420/consoleFull) for   PR 8381 at commit [`c4c1b28`](https://github.com/apache/spark/commit/c4c1b28685849b460ea343d8db3c8becce0d6de2).
 * This patch **fails Spark unit tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41420/
Test FAILed.

cloud-fan committed:
don't trim the top level Alias

Conflicts:
	sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

AmplabJenkins commented:
Merged build triggered.

AmplabJenkins commented:
Merged build started.

SparkQA commented:
[Test build #41430 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41430/consoleFull) for   PR 8381 at commit [`b2b8a01`](https://github.com/apache/spark/commit/b2b8a014fbfc6a567b3a0b0c9dc94979b08f42e5).
 * This patch merges cleanly.

SparkQA commented:
[Test build #41430 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41430/consoleFull) for   PR 8381 at commit [`b2b8a01`](https://github.com/apache/spark/commit/b2b8a014fbfc6a567b3a0b0c9dc94979b08f42e5).
 * This patch **passes all tests**.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test FAILed.

AmplabJenkins commented:
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/41430/
Test FAILed.
